### ISSUE TITLE
Metadata for ContentType and ContentEncoding Removed with copy_objects

### ISSUE THREAD

**Issue by juls858** [awslabs/aws-data-wrangler#827]
**Describe the bug**

Using [wr.copy_objects](https://aws-data-wrangler.readthedocs.io/en/stable/stubs/awswrangler.s3.copy_objects.html) erases metadata. Specifically ContentEncoding and ContenType system defined metadata are not copied.

Additionally specifiying _s3_additional_kwargs_ has no effect:
```python
wr.s3.copy_objects(
    paths=[file],
    source_path="s3://bucket/key/",
    target_path="s3://bucket/key/",
    s3_additional_kwargs=dict(
        ContentEncoding="gzip",
        ContentType="application/octet-stream",
        MetadataDirective="COPY",
    ),
)
```

**Environment**

Version 2.9.0

**To Reproduce**
```python
wr.s3.copy_objects(
    paths=[file],
    source_path="s3://bucket/key/",
    target_path="s3://bucket/key/"
)
```

**Comment by maxispeicher** [comment]
How exactly do you retrieve the metadata after copying? I've just uploaded a simple Hello World txt file with a `gzip` ContentEncoding and copied that file and for that single file the metadata was copied:
excerpt of source metadata:
```
...
   'content-encoding': 'gzip',
   'content-type': 'text/plain',
 'LastModified': datetime.datetime(2021, 7, 22, 10, 57, 23, tzinfo=tzutc()),
 'ContentEncoding': 'gzip',
 'ContentType': 'text/plain',
...
```

excerpt of target metadata:
```
...
   'content-encoding': 'gzip',
   'content-type': 'text/plain',
 'LastModified': datetime.datetime(2021, 7, 22, 10, 59, 26, tzinfo=tzutc()),
 'ContentEncoding': 'gzip',
 'ContentType': 'text/plain',
...
```

The used call for copying was simply:
```python
wr.s3.copy_objects(
  ["s3://bucket0/Hello.txt"], 
  source_path="s3://bucket0/", 
  target_path="s3://bucket1/")
```

**Comment by mas-dse-juremigi** [comment]
@maxispeicher. The issue is reproducible if you specify the parameter _s3_additional_kwargs_ as I have in the example.

I am using the following to check the metadata

```python
parts = urlparse(file)
bucket = parts.netloc
key = parts.path[1:]

s3.head_object(Bucket=bucket, Key=key)
```

I am using the keywords which work with the _copy_object_ method in [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.copy_object)

Specifying content encoding and content type work with boto3, but not with awswrangler.

**Comment by maxispeicher** [comment]
I can confirm that for `copy_objects` the additional kwargs that you specified are ignored. Only the kwargs in https://github.com/awslabs/aws-data-wrangler/blob/main/awswrangler/s3/_fs.py#L35 are accepted.

I think that we are talking about two separate things:

1. Do the target files have the same metadata as the source when using the default? (which is the case for me)
2. Should the user be able to overwrite the metadata during the copy process.

Regarding the first point it would be good to know if you can confirm my observation the original data is preserved, for the second point I'm generally open to including these kwargs but it might also be beneficial to include some more like `TaggingDirective`. WDYT @mas-dse-juremigi and @jaidisido?

**Comment by jaidisido** [comment]
I agree - the S3 kwargs list in [_fs](https://github.com/awslabs/aws-data-wrangler/blob/main/awswrangler/s3/_fs.py#L35) is out of date and would benefit from a refresh.

**Comment by maxispeicher** [comment]
Do you think it makes sense to have this check at all? It would also be possible to put the responsibility in the hands of the end user.

**Comment by mas-dse-juremigi** [comment]
@maxispeicher yes, please including TaggingDirective and/or MetadataDirective.   I think It should mimic the boto3 copy_object api if possible. There is no easy way to update file metadata except to copy the file over itself afaik.

**Comment by maxispeicher** [comment]
@mas-dse-juremigi 
With the merge of the linked PR the changes are available on the main branch. If you want to test it, you can do
```
pip uninstall awswrangler -y
pip install git+https://github.com/awslabs/aws-data-wrangler.git@main
```

`awsdatawrangler` should now forward all `s3_additional_kwargs` that are accepted by `botocore` to the corresponding functions.

**Comment by jaidisido** [comment]
Available in [2.11.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/2.11.0)
